envs/development/conda/fedbiomed-node-macosx.yaml

```diff
@@ -27,7 +27,7 @@ dependencies:
   - git ~=2.40.1
   - packaging ~=23.1
   # these two have to be aligned
-  - cryptography ~=40.0.0
```

Comment:
Hi @rtaiello @srcansiz 

Regarding the removal of `cryptography`:

Did we assess the compatibility of `pycryptodome` with `pyopenssl` ?

Touching to conda environments, especially when removing or changing version of packages requires much attention to avoid regression bugs :-/

 The version of `pyopenssl` we currently used 23.1.1 mentions compatibility
 with `cryptography <= 40.0.x`. What happens with `pycryptodome` ?

 The only use I found of `pyopenssl` is in `certificate_manager.py` am I missing something ?

We should probably at least:

1. test that we can re-generate the conda environments (remove them completely, then re-create from scratch) 
2. test that certificate manager works in this new setting


BTW I would avoid changing the version of `pyopenssl` (not done in the current PR, but if you would consider it ...). In this case we would need to redo quite intensive testing after removing/re-create from scratch all the conda envs in the CI. Also we would need pretty extensive manual testing with containers (no CI for that part yet).

Reply:
Hey @mvesin , as we discussed I just implemented the DH key and KA, with only `cryptography` so no new dependencies, we keep the same as before. many thx!

Reply:
Thank you @rtaiello !
Is there any specific reason why you change the versions of `cryptography` and `pyopenssl` to older versions than the one that are currently used ?

Reply:
hey @mvesin my mistake, I'll update them.